test/simulator/timestamp/src/timestamp_manager.cpp

```diff
@@ -42,52 +42,85 @@ timestamp_manager::get_timestamp_manager()
     return (_timestamp_manager_instance);
 }
 
-int
-timestamp_manager::validate_oldest_ts(uint64_t new_stable_ts, uint64_t new_oldest_ts)
+/*
+ * validate_oldest_and_stable_ts --
+ *     Validate both oldest and stable timestamps. - If stable and oldest were not passed in the
+ *     config, simply return. - Validation fails if Illegal timestamp value is passed (if less than
+ *     or equal to 0). - It is a no-op to set the oldest or stable timestamps behind the global
+ *     values. Hence, ignore and continue validating. - Ensure that the stable is greater than the
+ *     oldest timestamp.
+ */
+bool
+timestamp_manager::validate_oldest_and_stable_ts(
+  uint64_t &new_stable_ts, uint64_t &new_oldest_ts, bool &has_oldest, bool &has_stable)
```

Comment:
You don't need to pass those args by reference. You are just reading them. It's also dangerous to pass by reference without `const` as it makes modification possible.

Reply:
No this function changes these variables.